siteorigin-panels.php

```diff
@@ -353,33 +353,49 @@ public function generate_post_excerpt( $text ) {
 		// Check if this post has panels_data
 		$panels_data = get_post_meta( $post_id, 'panels_data', true );
 		if ( $panels_data && ! empty( $panels_data['widgets'] ) ) {
-			
-			foreach ( $panels_data['widgets'] as $widget ) {
+			$raw_excerpt = '';
+			$excerpt_length = apply_filters( 'excerpt_length', 55 );
+			foreach ( $panels_data['widgets'] as $i => $widget ) {
 				$panels_info = $widget['panels_info'];
 				if ( $panels_info['grid'] > 1 ) {
 					// Limiting search for a text type widget to the first two PB rows to avoid having excerpt content
 					// that's very far down in a post.
 					break;
 				}
 				if ( $panels_info['class'] == 'SiteOrigin_Widget_Editor_Widget' || $panels_info['class'] == 'WP_Widget_Text' ) {
-					$text = $widget['text'];
+					$raw_excerpt .= ( $i > 0 ? ' ' : '' ) . $widget['text'];
```

Comment:
Very minor, but if the first widget is a non text widget, then this ternary will add a space at the start of the next text widget's text. You could drop the ternary, just make it always add a space and call `trim` just before `wp_trim_words`. I haven't checked properly, but `wp_trim_words` probably does its own trimming, so that might not even be necessary.

Reply:
Hah, nice catch. Exactly what I was trying to avoid. :finnadie: